routing/turns_tts_text.cpp

```diff
@@ -62,11 +64,59 @@ string GetTtsText::GetTurnNotification(Notification const & notification) const
   if (notification.m_useThenInsteadOfDistance && notification.m_turnDir == CarDirection::None)
     return string();
 
-  string const dirStr = GetTextById(GetDirectionTextId(notification));
   if (dirStr.empty())
     return string();
 
-  string const distStr = GetTextById(GetDistanceTextId(notification));
+  std::string distStr = GetTextById(GetDistanceTextId(notification));
+
+  if (notification.m_distanceUnits > 0 && !notification.m_nextStreet.empty()) {
+    // We're going to pronounce the street name.
+
+    // Use UniString in order to operate on Unicode text
+    // First, let's get rid of unpronounceable symbols.
+    std::string streetOut = notification.m_nextStreet;
+    // Semicolons are between destinations
+    // and pronounced more like commas.
+    std::replace( streetOut.begin(), streetOut.end(), ';', ',');
+    // Open brackets have no pronunciation analogue
+    std::replace( streetOut.begin(), streetOut.end(), '[', ' ');
+    // Closed brackets end a highway and introduce the rest
+    std::replace( streetOut.begin(), streetOut.end(), ']', ':');
+    // An angle bracket is currently used to represent "to" a place
+    // Ideally we'd add a translation for "to" or create full format
+    // strings for better i18n support, like "turn onto %s toward %s"
+    std::replace( streetOut.begin(), streetOut.end(), '>', ',');
+
+    // Replace any full-stop characters to make TTS flow better.
+    // Full stops are: . (Period) or 。 (East Asian) or । (Hindi)
+    strings::ReplaceLast(distStr, ".", "");
+    strings::ReplaceLast(distStr, "。", "");
+    strings::ReplaceLast(distStr, "।", "");
+
+    // If the turn direction with the key +_street exists for this locale, use it (like make_a_right_turn_street)
+    string dirStreetStr = GetTextById(GetDirectionTextId(notification)+"_street");
+
+    if (!dirStreetStr.empty()) {
+      dirStr = dirStreetStr;
+    }
```

Comment:
dirStr = std::move(dirStreetStr); and without braces

Reply:
@vng these changes made in latest